### PR TITLE
Fix #869 & #870

### LINKED ISSUE #869: Datapack out of bounds
What are the steps to reproduce this issue?
-------------------------------------------
1. Silence someone via command, with time and reason
2. Unsilence him with reason
3. Silence was done but error in logs

What happens?
-------------
Command work but return error in logs

What were you expecting to happen?
----------------------------------
No datapack errors

Any logs, error output, etc.?
----------------------------

```
L 03/09/2023 - 20:54:25: SourceMod error session started
L 03/09/2023 - 20:54:25: Info (map "de_mirage") (file "/home//cs_go/csgo/addons/sourcemod/logs/errors_20230309.log")
L 03/09/2023 - 20:54:25: [SM] Exception reported: Invalid data pack position, 16 is out of bounds (6)
L 03/09/2023 - 20:54:25: [SM] Blaming: sbpp_comms.smx
L 03/09/2023 - 20:54:25: [SM] Call stack trace:
L 03/09/2023 - 20:54:25: [SM]   [0] DataPack.Position.set
L 03/09/2023 - 20:54:25: [SM]   [1] Line 1533, sbpp_comms.sp::Query_UnBlockSelect
L 03/09/2023 - 21:05:52: Error log file session closed.
```

Any other comments?
-------------------
Reported on discord by KA1ZER#0666
I was able to reproduct the issue on my server aswell.

What versions of software are you using?
----------------------------------------
**Operating System:** Ubuntu 20.04
**SourceBans++ Version:** 1.8.0
**SourceMod Version:** 1.11
**MetaMod Version:** 1.11

### PR DESCRIPTION
**Pull request by Rushaway**
## Description

- Fix the value of the by ReadCell instead.
- `g_bPlayerStatus` proper check

## Motivation and Context

Fix #869 & #870

## How Has This Been Tested?

- When retrying to reproduct #869 no more error and unblock work well.
- Wasn't able to have again false flag for #870

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: game/addons/sourcemod/scripting/sbpp_comms.sp [hunk 1]
```diff
@@ -1530,7 +1530,7 @@ public void Query_UnBlockSelect(Database db, DBResultSet results, const char[] e
 			{
 				// check result for possible combination with temp and time punishments (temp was skipped in code above)
 
-				dataPack.Position = view_as<DataPackPos>(16);
+				dataPack.Position = view_as<DataPackPos>(6);
```
Comment (rumblefrog) [inline game/addons/sourcemod/scripting/sbpp_comms.sp]: I believe we should resolve the issue of using view_as<DataPackPos>; DataPackPos is never meant to be used this way

They are references to SP internals, which could be wrong someday; we should start reading cells to get to the proper position.

**Review by rumblefrog — CHANGES_REQUESTED** [review]

File: game/addons/sourcemod/scripting/sbpp_comms.sp (hunk 1 above)
Comment (Hackmastr) [reply game/addons/sourcemod/scripting/sbpp_comms.sp]: @Rushaway any update on this?

Comment (Rushaway) [reply game/addons/sourcemod/scripting/sbpp_comms.sp:1533]: I will try to get a deeper look on it this week. I'm bus as hell these week.

File: game/addons/sourcemod/scripting/sbpp_comms.sp [hunk 2]
```diff
@@ -1530,7 +1536,8 @@ public void Query_UnBlockSelect(Database db, DBResultSet results, const char[] e
 			{
 				// check result for possible combination with temp and time punishments (temp was skipped in code above)
 
-				dataPack.Position = view_as<DataPackPos>(16);
+				dataPack.Reset();
+				dataPack.ReadCell();
```
Comment (rumblefrog) [inline game/addons/sourcemod/scripting/sbpp_comms.sp:1540]: Sanity check here, is a single readcell actually skipping to 16?

File: game/addons/sourcemod/scripting/sbpp_comms.sp [hunk 3]
```diff
@@ -290,13 +290,19 @@ public void OnClientPostAdminCheck(int client)
...
-	if (clientAuth[0] == 'B' || clientAuth[9] == 'L' || !DB_Connect())
+	/* Can't connect to DB */
+	if (!DB_Connect())
 	{
 		g_bPlayerStatus[client] = true;
 		return;
 	}
 
+	/*  Do not check bots or player returning as Steamid as Pending | Stop_ignoring_retvals */
+	if (strncmp(clientAuth[6], "ID_", 3) != -1 || clientAuth[0] == 'B')
```
Comment (DosMike) [inline game/addons/sourcemod/scripting/sbpp_comms.sp]: Shouldn't the [`strncmp`](https://sourcemod.dev/#/string/function.strncmp)s (across the PR) compare to `== 0` to test for equality? This is not [`StrContains`](https://sourcemod.dev/#/string/function.StrContains), which returns -1 if the search string is not contained.

**Comment by Rushaway** [comment]
@cruze03 You was reading cell to the second position instead of 3.

**Pull request closed without merging**